src/cargo/core/features.rs

```diff
@@ -186,6 +186,7 @@ features! {
         [stable] rename_dependency: bool,
 
         // Whether a lock file is published with this crate
+        // This is deprecated, and will likely be removed in a future version.
```

Comment:
Why wasnt this marked as stable instead?

Reply:
We did not have a compelling enough reason to add a mechanism to disable including the lock file.

Reply:
I dont think it is possible to disable a stable feature. There is no `remove` method, while there is an `add` method:

https://github.com/ehuss/cargo/blob/34307c6122eb1b9368c70f3b3ca3900ef08ad6b8/src/cargo/core/features.rs#L224

Reply:
I'm not sure I understand.  We didn't feel like it was important to allow someone to specify `publish-lockfile = false` in `Cargo.toml`, so we plan to just completely remove that part.  I could have removed it immediately, but I figured I'd leave it in for a while in case we changed our mind.

Reply:
https://github.com/rust-lang/cargo/blob/34307c6122eb1b9368c70f3b3ca3900ef08ad6b8/src/cargo/core/features.rs#L115-L128

When a feature is marked stable, it is always enabled.

Reply:
The feature is, but not the `publish-lockfile` flag in the manifest. They two are independent.  In this case, we don't want either of them (the feature or the flag).

Reply:
I get it.